app-crypt/johntheripper/johntheripper-1.8.0.recipe

```diff
@@ -0,0 +1,178 @@
+SUMMARY="A fast password cracker"
+DESCRIPTION="A fast password cracker, currently available for many flavors of \
+Unix, Windows, DOS, and OpenVMS. Its primary purpose is to detect weak Unix \
+passwords. Besides several crypt(3) password hash types most commonly found on \
+various Unix systems, supported out of the box are Windows LM hashes, plus \
+lots of other hashes and ciphers in the community-enhanced version."
+HOMEPAGE="http://www.openwall.com/john"
+COPYRIGHT="1996-2013 Solar Designer
+	2014 John the Ripper '-jumbo' authors"
+LICENSE="GNU GPL v2
+	John the Ripper"
+REVISION="1"
+jumboVersion="1"
+SOURCE_URI="http://www.openwall.com/john/j/john-$portVersion-jumbo-$jumboVersion.tar.xz"
+CHECKSUM_SHA256="bac93d025995a051f055adbd7ce2f1975676cac6c74a6c7a3ee4cfdd9c160923"
+SOURCE_DIR="john-$portVersion-jumbo-$jumboVersion"
+# Upstream patches, can be dropped in the next version
+PATCHES="johntheripper-$portVersion-upstreamed.patchset
+	johntheripper-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 !x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	johntheripper$secondaryArchSuffix = $portVersion
+	cmd:john = $portVersion
+	"
+
+bins=(calc_stat cprepair genmkvpwd luks2john mkvcalcproba raw2dyna
+	tgtsnarf uaf2john wpapcap2john)
+syms=(base64conv dmg2john gpg2john hccap2john keepass2john keychain2john
+	keyring2john keystore2john kwallet2john pfx2john putty2john pwsafe2john
+	racf2john rar2john ssh2john truecrypt_volume2john unafs undrop unique
+	unshadow zip2john)
+for b in ${bins[@]} ${syms[@]}
+do
+	PROVIDES+="
+		cmd:$b = $portVersion
+		"
+done
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcrypto$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+GLOBAL_WRITABLE_FILES="
+	settings/john/john.conf keep-old
+	settings/john/john.ini keep-old
+	"
+USER_SETTINGS_FILES="
+	settings/john directory
+	settings/john/john.local.conf template $relativeDocDir/john.local.conf.example
+	"
+
+BUILD()
+{
+	local johnFlags=(
+		"-DJOHN_SYSTEMWIDE=1"
+		"-DJOHN_SYSTEMWIDE_EXEC=\"\\\"$libExecDir/john\\\"\""
+		"-DJOHN_SYSTEMWIDE_HOME=\"\\\"$dataDir/john\\\"\""
+		"-DJOHN_PRIVATE_HOME=\"\\\"~data/john\\\"\""
+		"-DCFG_FULL_NAME=\"\\\"$settingsDir/john/john.cfg\\\"\""
+		"-DCFG_ALT_NAME=\"\\\"$settingsDir/john/john.ini\\\"\""
```

Comment:
**WARNING:** this comment might be irrelevant.
* I don't know if the first change in this diff is a good idea. In case you think it is, then it could be nice to do the same change on the other lines too.
* For the 2 other changes, regarding the definitions of CFG_FULL_NAME and CFG_ALT_NAME, I don't know if they are required or not. I have to admit that I don't know if john can access /boot/system/setting/john/ using this friendly path or if it has to use the `$settingDir/john/` path.

```diff
 BUILD()
 {
+	local constantSystemSettingsDir="`finddir B_SYSTEM_SETTINGS_DIRECTORY`"
 	local johnFlags=(
 		"-DJOHN_SYSTEMWIDE=1"
-		"-DJOHN_SYSTEMWIDE_EXEC=\"\\\"$libExecDir/john\\\"\""
+		'-DJOHN_SYSTEMWIDE_EXEC="\"'"$libExecDir"'/john\""'
 		"-DJOHN_SYSTEMWIDE_HOME=\"\\\"$dataDir/john\\\"\""
 		"-DJOHN_PRIVATE_HOME=\"\\\"~data/john\\\"\""
-		"-DCFG_FULL_NAME=\"\\\"$settingsDir/john/john.cfg\\\"\""
+		"-DCFG_FULL_NAME=\"\\\"$constantSystemSettingsDir/john/john.cfg\\\"\""
-		"-DCFG_ALT_NAME=\"\\\"$settingsDir/john/john.ini\\\"\""
+		"-DCFG_ALT_NAME=\"\\\"$constantSystemSettingsDir/john/john.ini\\\"\""
```

Reply:
I prefer my quoting solution more as mixing quotes is prone to error, especially when I'm unable to phrase it